### PR TITLE
Task/food categories

### PR DESCRIPTION
**Pull request by delbertlegg**
Addresssing Issues #27 and #22. Added endpoint to get all food taxonomies (`/food`). Also added a service layer to decouple logic from routes, and added dependency injection so services can be quickly switched out when changing/adding data sources.

In order to change services, add a folder under the `services` path and create a service for each route (agency, location, taxonomy). Use the same naming conventions as the current services (`getAll`, `get`, and other route specific function names). Then use those services by changing the `getRoutes` method in `server.js` using the path you just added.

### CODE REVIEW

File: api/controllers/routes/v2/taxonomy.js [hunk 1]
```diff
@@ -1,50 +1,44 @@
...
+     */
+    app.get("/", function (req, res) {
+        service.getAll(req, res);
+    });
+
+    /**
+     * Get Request for all Food categories
+     */
+    app.get("/food", function (req, res) {
+        service.getFoodCategories(req, res);
```
Comment (vijayyadav06) [inline api/controllers/routes/v2/taxonomy.js]: wondering if we can method as getAllFoodCategories!

File: server.js [hunk 2]
```diff
@@ -13,17 +13,11 @@ const taxonomyRouterV1 = require('./api/controllers/routes/v1/taxonomy');
...
-const locationRouterV2 = require('./api/controllers/routes/v2/location');
-
 app.use('/api/v1/taxonomy', taxonomyRouterV1);
 app.use('/api/v1/agency', agencyRouterV1);
 app.use('/api/v1/location', locationRouterV1);
 
-app.use('/api/v2/taxonomy', taxonomyRouterV2);
-app.use('/api/v2/agency', agencyRouterV2);
-app.use('/api/v2/location', locationRouterV2);
+app.use(require('./api/routes.js').getRoutes('v2', 'ckan'));
```
Comment (vijayyadav06) [inline server.js]: this is cool, allowing us to pass version and platform as an argument to route.

File: api/controllers/services/ckan/taxonomyService.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        this.requestUtils.getList(queryString, res, Taxonomy.getList);
+    }
+
+    get(req, res) {
+        const queryString = this.uri + this.queryUtils.setDefaultFilters(`"TAXON_ID" = ${req.params.id}`, this.tableName);
+        this.requestUtils.getObject(queryString, res, Taxonomy.getObject);
+    }
+
+    getFoodCategories(req, res) {
+        const queryString = this.uri + this.queryUtils.setDefaultFilters('"TAXONOMY_CODE" LIKE N\'BD-%\'', this.tableName);
```
Comment (vijayyadav06) [inline api/controllers/services/ckan/taxonomyService.js]: nice this was easy. didn't know that Food subcategories all were starting as BD-

File: api/controllers/routes/v2/taxonomy.js (hunk 1 above)
Comment (delbertlegg) [reply api/controllers/routes/v2/taxonomy.js]: Yep, absolutely...updated method name.

File: server.js (hunk 2 above)
Comment (delbertlegg) [reply server.js]: Thanks...this should make it easier to inject mock services for testing and change data services quickly for memcaching or 2.0 of SCOS.

File: api/controllers/services/ckan/taxonomyService.js (hunk 3 above)
Comment (delbertlegg) [reply api/controllers/services/ckan/taxonomyService.js]: Yep! At least with the current data set :)

**Pull request merged**